sycl/include/sycl/ext/oneapi/matrix/matrix-hip.hpp

```diff
@@ -0,0 +1,389 @@
+
+//===-------- matrix-hip.hpp - matrix ext impl ---*- C++ -*-------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===-------------------------------------------------------------------=== //
+
+#pragma once
+#include "matrix-unified-utils.hpp"
+#include <sycl/ext/oneapi/bfloat16.hpp>
+
+#if defined(__gfx90a__)
+#define __HIP_PLATFORM_AMD_MFMA__
+#endif
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+namespace matrix {} // namespace matrix
+} // namespace experimental
+
+using matrix_layout = sycl::ext::oneapi::experimental::matrix::layout;
+using matrix_use = sycl::ext::oneapi::experimental::matrix::use;
+
+namespace detail {
+
+template <typename T, matrix_use Use, size_t Rows, size_t Cols,
+          matrix_layout Layout = matrix_layout::dynamic, typename Cond = void>
+struct joint_matrix_hip;
+
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__HIP_PLATFORM_AMD_MFMA__)
+
+template <typename T> struct to_hip_type {
+  using type = T;
+};
+
+template <> struct to_hip_type<bfloat16> {
+  using type = __bf16;
```

Comment:
Interesting. I guess this means that in the amdgcn backend test-e2e cases are correctly casting from `sycl::bfloat16` to `__bf16`. Although I do see that the mfma test etc is using integer values for bfloat16 cases also. It might be worth just double checking that "real numbers"/float values can be used as input and the correct values are preserved when casting from `sycl::bfloat16` to `__bf16`. I guess you maybe did this already though.